packages/compass-indexes/src/modules/search-indexes.spec.ts

```diff
@@ -269,6 +272,96 @@ describe('search-indexes module', function () {
       expect(store.getState().searchIndexes.error).to.be.undefined;
     });
   });
+
+  describe('openSearchIndexes and closeSearchIndexes', function () {
+    let clock: sinon.SinonFakeTimers;
+
+    before(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    after(() => {
+      clock.restore();
+    });
+
+    it('starts and stops the polling', async function () {
+      const pollInterval = 5000;
+
+      const getSearchIndexesStub = sinon.stub().resolves(searchIndexes);
+      const store = setupStore(
+        {
+          isSearchIndexesSupported: true,
+        },
+        {
+          getSearchIndexes: getSearchIndexesStub,
+        }
+      );
+
+      const waitForStatus = async (status: FetchStatus) => {
+        while (store.getState().searchIndexes.status !== status) {
+          /*
+          This looks like a pretty tight loop, but the await at least gives the
+          stub time to resolve. If we wait using a timeout it will never fire
+          because this test takes control of the clock. In fact this while
+          condition is never false when the test passes so it never enters the
+          loop because the await on this function itself gives the stub time to
+          execute.
+          */
+          await Promise.resolve();
+        }
+      };
```

Comment:
Don't have a strong preference for or against writing a dedicated helper, but we usually just use `waitFor` from the testing-library for stuff like this instead of writing our own

Reply:
I'll see if that works. If it does any kind of timeout or interval internally it will hang because of the comment above.